src/API/Site/Controllers/Assets/AssetSuggestionsController.php

```diff
@@ -0,0 +1,162 @@
+ */
+class AssetSuggestionsController extends BaseController {
+
+	/**
+	 * Service used to populate ads suggestions data.
+	 *
+	 * @var AssetSuggestionsService
+	 */
+	protected $asset_suggestions_service;
+
+	/**
+	 * AssetSuggestionsController constructor.
+	 *
+	 * @param RESTServer              $server
+	 * @param AssetSuggestionsService $asset_suggestion
+	 */
+	public function __construct( RESTServer $server, AssetSuggestionsService $asset_suggestion ) {
+		parent::__construct( $server );
+		$this->asset_suggestions_service = $asset_suggestion;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'assets/final-url/suggestions',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_final_urls_suggestions_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_collection_params(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params(): array {
+		return [
+			'search'   => [
+				'description'       => __( 'Search for post title or term name', 'google-listings-and-ads' ),
+				'type'              => 'string',
+				'default'           => '',
+				'sanitize_callback' => 'sanitize_text_field',
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+			'per_page' => [
+				'description'       => __( 'The number of items to be return', 'google-listings-and-ads' ),
+				'type'              => 'number',
+				'default'           => 30,
+				'sanitize_callback' => 'absint',
+				'minimum'           => 1,
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+			'order_by' => [
+				'description'       => __( 'Sort retrieved items by parameter', 'google-listings-and-ads' ),
+				'type'              => 'string',
+				'default'           => 'title',
+				'sanitize_callback' => 'sanitize_text_field',
+				'enum'              => [ 'id', 'type', 'title', 'url' ],
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+		];
+	}
+
+	/**
+	 * Get the callback function for the list of final-url suggestions request.
+	 *
+	 * @return callable
+	 */
+	protected function get_final_urls_suggestions_callback(): callable {
+		return function( Request $request ) {
+			$search   = $request->get_param( 'search' );
+			$per_page = $request->get_param( 'per_page' );
+			$order_by = $request->get_param( 'order_by' );
+			return array_map(
+				function( $item ) use ( $request ) {
+					$data = $this->prepare_item_for_response( $item, $request );
+					return $this->prepare_response_for_collection( $data );
+				},
+				$this->asset_suggestions_service->get_final_urls_suggestions( $search, $per_page, $order_by )
```

Comment:
💅  minor

Same as in `https://github.com/woocommerce/google-listings-and-ads/pull/1784/files#r1031025484` there are some plural and singular inconsistencies